### ISSUE TITLE
Support win. system build source code

### ISSUE THREAD

**Issue by RaidOpe** [medialab/xsv#109]
I tried to use branches on the win system, but it seems that some libraries do not support win.

**Comment by RaidOpe** [comment]
```
error[E0433]: failed to resolve: could not find `unix` in `os`
 --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pager-0.16.1\src\utils.rs:2:14
  |
2 | use std::os::unix::ffi::{OsStrExt, OsStringExt};
  |              ^^^^ could not find `unix` in `os`

error[E0412]: cannot find type `pid_t` in crate `libc`
  --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pager-0.16.1\src\utils.rs:17:31
   |
17 | pub(crate) fn fork() -> libc::pid_t {
   |                               ^^^^^ not found in `libc`

error[E0425]: cannot find function `fork` in crate `libc`
  --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pager-0.16.1\src\utils.rs:18:20
   |
18 |     unsafe { libc::fork() }
   |                    ^^^^ not found in `libc`

error[E0425]: cannot find value `STDOUT_FILENO` in crate `libc`
   --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pager-0.16.1\src\lib.rs:202:55
    |
202 |         if self.skip_on_notty && !utils::isatty(libc::STDOUT_FILENO) {
    |                                                       ^^^^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find value `STDOUT_FILENO` in crate `libc`
   --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pager-0.16.1\src\lib.rs:218:52
    |
218 |                     utils::dup2(main_stdout, libc::STDOUT_FILENO);
    |                                                    ^^^^^^^^^^^^^ not found in `libc`

error[E0425]: cannot find value `STDIN_FILENO` in crate `libc`
   --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pager-0.16.1\src\lib.rs:223:52
    |
223 |                     utils::dup2(pager_stdin, libc::STDIN_FILENO);
    |                                                    ^^^^^^^^^^^^ not found in `libc`

error[E0599]: no method named `into_vec` found for struct `OsString` in the current scope
 --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pager-0.16.1\src\utils.rs:6:44
  |
6 |     unsafe { CString::from_vec_unchecked(s.into_vec()) }
  |                                            ^^^^^^^^ method not found in `OsString`

error[E0599]: no method named `as_bytes` found for reference `&OsString` in the current scope
  --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pager-0.16.1\src\utils.rs:33:35
   |
33 |         unsafe { libc::putenv(env.as_bytes().as_ptr() as *mut _) };
   |                                   ^^^^^^^^ help: there is a method with a similar name: `as_encoded_bytes`

error[E0061]: this function takes 3 arguments but 1 argument was supplied
   --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\pager-0.16.1\src\utils.rs:50:25
    |
50  |     assert_eq!(unsafe { libc::pipe(fds.as_mut_ptr()) }, 0);
    |                         ^^^^^^^^^^------------------ two arguments of type `u32` and `i32` are missing
    |
note: function defined here
   --> C:\Users\77456\.cargo\registry\src\index.crates.io-6f17d22bba15001f\libc-0.2.148\src\windows\mod.rs:499:12
    |
499 |     pub fn pipe(fds: *mut ::c_int, psize: ::c_uint, textmode: ::c_int) -> ::c_int;
    |            ^^^^
help: provide the arguments
    |
50  |     assert_eq!(unsafe { libc::pipe(fds.as_mut_ptr(), /* u32 */, /* i32 */) }, 0);
```

**Comment by Yomguithereal** [comment]
Ok I see. Seems to be the `pager` library. I can add some windows conditionals to the compilation to avoid the issue. Which means the `-p` flag will not work for the `view` command unfortunately but I don't know any workaround and I don't have access to a windows system to perform tests on my end.

**Comment by Yomguithereal** [comment]
@RaidOpe I just pushed a commit that should take care of the optional dependency and should ensure that the tool compiles on windows albeit without a working view -p flag. Could you report if this fixes your problem?

**Comment by RaidOpe** [comment]
latest build result
```
error[E0433]: failed to resolve: could not find `unix` in `os`
 --> src\cmd\foreach.rs:6:14
  |
6 | use std::os::unix::ffi::OsStrExt;
  |              ^^^^ could not find `unix` in `os`

warning: unused import: `ColumIndexationBy`
  --> src\xan\mod.rs:12:22
   |
12 | pub use xan::types::{ColumIndexationBy, DynamicValue};
   |                      ^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error[E0599]: no function or associated item named `from_bytes` found for struct `OsStr` in the current scope
  --> src\cmd\foreach.rs:83:30
   |
83 |         let command = OsStr::from_bytes(&templated_command);
   |                              ^^^^^^^^^^ function or associated item not found in `OsStr`

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
warning: `xsv` (bin "xsv") generated 1 warning
error: could not compile `xsv` (bin "xsv") due to 2 previous errors; 1 warning emitted
```

**Comment by Yomguithereal** [comment]
Fair enough, now we have an issue with the `foreach` command (which admittedly is not really windows compatible). Let me fix that.

**Comment by Yomguithereal** [comment]
Just committed something to shunt the foreach command on windows. Can you try again please?

**Comment by RaidOpe** [comment]
I was also surprised
```
error[E0433]: failed to resolve: could not find `unix` in `os`
 --> src\cmd\foreach.rs:6:14
  |
6 | use std::os::unix::ffi::OsStrExt;
  |              ^^^^ could not find `unix` in `os`

error[E0599]: no function or associated item named `from_bytes` found for struct `OsStr` in the current scope
  --> src\cmd\foreach.rs:83:30
   |
83 |         let command = OsStr::from_bytes(&templated_command);
   |                              ^^^^^^^^^^ function or associated item not found in `OsStr`

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `xsv` (bin "xsv") due to 2 previous errors
```

**Comment by Yomguithereal** [comment]
Ok, new commit with an attempt to fix this. Sorry for the to and fro but I don't have a computer with windows and this is a bit new to me. Tell me if this finally fixes it :)

**Comment by RaidOpe** [comment]
No, no, it would be nice to get a positive response. I'm happy to participate in the debugging.

Good news, the build was successful!

`Finished release [optimized + debuginfo] target(s) in 1m 20s`

The build time of the 2.7Ghz, 16-thread processor is really fast, though

Cheer! Thank you for your great work. I tried the command and it works fine. Even if I don’t eat honey, dopamine is secreted like crazy. I wish you a good day. If possible, I can also donate a little，Of course, thanks to the contribution of the master branch，too.  Although my income is meager

**Comment by Yomguithereal** [comment]
Can I ask you what lead you to our fork?

**Comment by RaidOpe** [comment]
I began to learn about python's csv tool csvkit, mainly because csv is used as a transitional analysis tool, but the commands of this tool appear. . . Somewhat slow. I also think it's great for temporary use. Then github recommended the xsv of the main branch today, but compared to csvkit, the functions are still a bit less, but I appreciate the superior features of the rust tool, so I looked to see if Fork has developed more functions. So, pop. Maybe python's out-of-the-box functionality attracts too many people. Feeling lucky now.
